### PR TITLE
qmltermwidget: 2018-11-24 -> unstable-2022-01-09

### PR DESCRIPTION
**Pull request by GaetanLepage**
###### Description of changes

Update `qmltermwidget` to the latest git version.
This will hopefully fix `cool-retro-term`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/qmltermwidget/default.nix [hunk 1]
```diff
@@ -1,25 +1,45 @@
...
-  pname = "qmltermwidget-unstable";
+  version = "unstable-2022-01-09";
+  pname = "qmltermwidget";
 
   src = fetchFromGitHub {
     repo = "qmltermwidget";
     owner = "Swordfish90";
-    rev = "48274c75660e28d44af7c195e79accdf1bd44963";
-    sha256 = "028nb1xp84jmakif5mmzx52q3rsjwckw27jdpahyaqw7j7i5znq6";
+    # rev = "refs/tags/${version}";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qmltermwidget/default.nix:19]:
```suggestion
```

File: pkgs/development/libraries/qmltermwidget/default.nix [hunk 2]
```diff
@@ -1,25 +1,45 @@
...
   };
 
-  buildInputs = [ qtbase qtquick1 qtmultimedia ]
-                ++ lib.optional stdenv.isDarwin utmp;
+  buildInputs = [
+    qtbase
+    qtquick1
+    qtmultimedia
+  ]
+  ++ lib.optional stdenv.isDarwin utmp;
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qmltermwidget/default.nix:29]:
```suggestion
  ] ++ lib.optional stdenv.isDarwin utmp;
```

File: pkgs/development/libraries/qmltermwidget/default.nix [hunk 3]
```diff
@@ -28,7 +48,12 @@ stdenv.mkDerivation {
       --replace '$$[QT_INSTALL_QML]' "/$qtQmlPrefix/"
   '';
 
-  installFlags = [ "INSTALL_ROOT=$(out)" ];
+  installFlags = [ "INSTALL_ROOT=${placeholder "out"}" ];
+
+  preInstall = ''
+    mkdir -p $out/lib/qt-5.15.9/qml//QMLTermWidget/kb-layouts/
+    touch $out/lib/qt-5.15.9/qml//QMLTermWidget/kb-layouts/macbook.keytab
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qmltermwidget/default.nix:55]: The version number must be substituted

Comment (GaetanLepage) [reply pkgs/development/libraries/qmltermwidget/default.nix:55]: This was just an experiment to debug the install phase.

**Comment by GaetanLepage** [comment]
@SuperSandro2000 I was able to fix the build.
Also, I tested with `cool-retro-term` and it gets indeed fixed thanks to this PR.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 244503` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>calyx-vpn</li>
    <li>cool-retro-term</li>
    <li>libsForQt5.qmlkonsole</li>
    <li>libsForQt5.qmltermwidget</li>
    <li>libsForQt5.station</li>
    <li>riseup-vpn</li>
  </ul>
</details>

File: pkgs/development/libraries/qmltermwidget/default.nix [hunk 4]
```diff
@@ -1,34 +1,43 @@
...
+, qmake
+, qtmultimedia
+, utmp
+}:
 
 stdenv.mkDerivation {
-  version = "2018-11-24";
-  pname = "qmltermwidget-unstable";
+  version = "unstable-2022-01-09";
+  pname = "qmltermwidget";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qmltermwidget/default.nix:13]: I would switch this but not important

File: pkgs/development/libraries/qmltermwidget/default.nix [hunk 5]
```diff
@@ -1,34 +1,43 @@
...
 
-  buildInputs = [ qtbase qtquick1 qtmultimedia ]
-                ++ lib.optional stdenv.isDarwin utmp;
+  buildInputs = [
+    qtbase
+    qtquick1
+    qtmultimedia
+  ] ++ lib.optional stdenv.isDarwin utmp;
+
   nativeBuildInputs = [ qmake ];
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qmltermwidget/default.nix:28]: also those two

**Review by SuperSandro2000 — APPROVED** [review]

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).

Result of `nixpkgs-review pr 244503` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
6 packages built:
</summary>
<ul>
<li>calyx-vpn</li>
<li>cool-retro-term</li>
<li>libsForQt5.qmlkonsole (plasma5Packages.qmlkonsole)</li>
<li>libsForQt5.qmltermwidget (plasma5Packages.qmltermwidget)</li>
<li>libsForQt5.station (plasma5Packages.station)</li>
<li>riseup-vpn</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
calyx-vpn:
<details>

warning: EvalError
Cannot evaluate attribute ‘calyx-vpn’ in ‘/nix/store/3k4vy644vlk8d29shdbf66164hqfjm0k-nixpkgs’.
</details>
cool-retro-term:
<details>

warning: EvalError
Cannot evaluate attribute ‘cool-retro-term’ in ‘/nix/store/3k4vy644vlk8d29shdbf66164hqfjm0k-nixpkgs’.
</details>
libsForQt5.qmlkonsole:
<details>

warning: EvalError
Cannot evaluate attribute ‘libsForQt5.qmlkonsole’ in ‘/nix/store/3k4vy644vlk8d29shdbf66164hqfjm0k-nixpkgs’.
</details>
libsForQt5.qmltermwidget:
<details>

warning: EvalError
Cannot evaluate attribute ‘libsForQt5.qmltermwidget’ in ‘/nix/store/3k4vy644vlk8d29shdbf66164hqfjm0k-nixpkgs’.
</details>
libsForQt5.station:
<details>

warning: EvalError
Cannot evaluate attribute ‘libsForQt5.station’ in ‘/nix/store/3k4vy644vlk8d29shdbf66164hqfjm0k-nixpkgs’.
</details>
riseup-vpn:
<details>

warning: EvalError
Cannot evaluate attribute ‘riseup-vpn’ in ‘/nix/store/3k4vy644vlk8d29shdbf66164hqfjm0k-nixpkgs’.
</details>
</details>

**Comment by SuperSandro2000** [comment]
By saying it is not important I wanted to imply that you just leave it like it is and I merge the PR :)

**Pull request merged**